File: DSA/HR/plus_minus.java
import java.util.List;

class Result {
    public static void plusMinus(List<Integer> arr) {
        int neg = 0, pos = 0, zero = 0;
        for (int num : arr) {
            if (num == 0) {
                zero++;
            } else if (num < 0) {
                neg++;
            } else {
                pos++;
            }
        }
        int total = neg + pos + zero;
        System.out.printf("%.6f\n", (float) pos / total);
        System.out.printf("%.6f\n", (float) neg / total);
        System.out.printf("%.6f\n", (float) zero / total);
    }
}

File: DSA/HR/plus_minus.cpp
#include <vector>
#include <iostream>
using namespace std;

void plusMinus(vector<int> arr)
{
    int neg{}, zero{}, pos{}, count{};

    for (int num : arr)
    {
        if (num < 0)
            neg++;
        else if (num == 0)
            zero++;
        else
            pos++;
        count++;
    }

    std::cout << (double)pos / count << "\n";
    std::cout << (double)neg / count << "\n";
    std::cout << (double)zero / count << "\n";
}